test/CardTest.java

```diff
@@ -0,0 +1,63 @@
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Created by marti on 2017.05.22..
+ */
+public class CardTest {
+
+    private Card card;
+
+    @Test
+    public void testIsCardColorWrong() {
+        assertThrows(IllegalArgumentException.class, () ->
+        {
+            new Card("W5");
+        });
+    }
+
+    @Test
+    public void testIsCardValueLarger() {
+        assertThrows(IllegalArgumentException.class, () ->
+        {
+            new Card("S11");
+        });
+    }
+
+    @Test
+    public void testIsCardValueSmaller() {
+        assertThrows(IllegalArgumentException.class, () ->
+        {
+            new Card("S1");
+        });
+    }
+
+    @Test
+    public void testIsCardValueNotValidLetter() {
+        assertThrows(IllegalArgumentException.class, () ->
+        {
+            new Card("SP");
+        });
+    }
+
+    @Test
+    public void testCardValuesAsLettersByGetValue() {
+        char[] cardLetters = new char[] {'j','q','k','a'};
+        List<Card> cards = new ArrayList<>();
+        List<Integer> cardValues = new ArrayList<>();
+        IntStream.range(0, 4).forEach(i ->cards.add(new Card("h" + cardLetters[i])));
```

Comment:
Ugyi vagy, hogy elkezdtel stream-ezni. Viszont a legtobb esetben nincs szukseg `new ArrayList`meg hasonlokra, mert a stream alapbol meg tuda csinalni. Most pl:
```
IntStream.range(0,4)
    .map(i -> new Card("h" + cardLetters[i]))
    .map(card::getValue)
    .collect(Collectors.toList())
```
es kesz is a cardValues.